packages/kbn-securitysolution-io-ts-types/src/time_duration/index.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import * as t from 'io-ts';
+import { Either } from 'fp-ts/lib/Either';
+
+/**
+ * Types the TimeDuration as:
+ *   - A string that is not empty, and composed of a positive integer greater than 0 followed by a unit of time
+ *   - in the format {safe_integer}{timeUnit}, e.g. "30s", "1m", "2h"
+ */
+export const TimeDuration = new t.Type<string, string, unknown>(
```

Comment:
I think it's not generic enough to rename it so, since Date Math also allows for expressions such as `/d`, but I could expand it to cover for the other time units of day, weeks, month and year. WDYT?

Reply:
@jpdjere I think the implementation we ended up having in https://github.com/elastic/kibana/pull/140626 is good. No need for renaming to `DateMathDuration`.